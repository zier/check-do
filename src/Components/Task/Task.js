import React from 'react';
import PropTypes from 'prop-types';

import { Card, Tag, Icon  } from 'antd';

import './Task.css'

const Task = ({ id, title, description, isDone, tags, onClickToggleDone, onClickDelete }) => {
  // region Event: Buttons
  const onClickToggleTask = () => {
    onClickToggleDone(id)
  }

  const onClickDeleteTask = () => {
    onClickDelete(id)
  }
  // endregion

  let textLineThrough = { textDecoration: 'none' }
  if (isDone) {
    textLineThrough.textDecoration = 'line-through';
  }

  return (
    <Card style={{ padding: '0px' }} bordered={true}>
      <div className="control-buttons">
        <Icon onClick={onClickToggleTask} className="checkButton" type={isDone ? 'check-circle' : 'check-circle-o'} />
        <Icon onClick={onClickDeleteTask} className="closeButton" type="close" />
      </div>
      <h4 style={textLineThrough}>{title}</h4>
      <p style={textLineThrough}>{description}</p>
      {tags.map(tag => <Tag key={tag} className="taskTag">{tag}</Tag>)}
    </Card>
  )
}

Task.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  isDone: PropTypes.bool,
  tags: PropTypes.array.isRequired,
  onClickToggleDone: PropTypes.func,
  onClickDelete: PropTypes.func
}

export default Task;
