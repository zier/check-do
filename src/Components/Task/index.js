import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Card } from 'antd';

class Task extends Component {
  render() {
    const { title, description, onClick } = this.props

    return (
      <Card onClick={onClick} title={title} bordered={true}>
        {description}
      </Card>
    );
  }
}

Task.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  onClick: PropTypes.func
};

export default Task;
