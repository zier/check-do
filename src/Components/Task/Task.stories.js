import React from 'react';
import Task from '../Task';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

export const task = {
  title: 'Go home',
  description: 'go to home after meeting',
};

export const actions = {
  onClick: action('onClickTask'),
};


storiesOf('Task', module)
  .addDecorator(story => <div style={{ padding: '3rem' }}>{story()}</div>)
  .add('with data', () => <Task {...task} {...actions} />)
