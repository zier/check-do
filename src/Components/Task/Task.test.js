import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Task from './Task.js'

describe('Task work correctly', () => {
  const props = {
    title: "title-test",
    description: "description-test",
    isDone: false,
  }

  const actions = {
    onClickToggleDone: jest.fn(),
    onClickDelete: jest.fn(),
  }

  const wrapper = shallow(<Task {...props} {...actions}/>)

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('props is work', () => {
    expect(wrapper.find('h4').text()).toBe(props.title)
    expect(wrapper.find('p').text()).toBe(props.description)
  })

  it('actions is work', () => {
    wrapper.find('.checkButton').simulate('click')
    wrapper.find('.closeButton').simulate('click')

    expect(actions.onClickToggleDone.mock.calls.length).toBe(1)
    expect(actions.onClickDelete.mock.calls.length).toBe(1)
  })

  it('task is Done', () => {
    wrapper.setProps({ isDone: false })
    expect(wrapper.find('h4')).toHaveStyle({ textDecoration: 'none' })
    expect(wrapper.find('p')).toHaveStyle({ textDecoration: 'none' })

    wrapper.setProps({ isDone: true })
    expect(wrapper.find('h4')).toHaveStyle({ textDecoration: 'line-through' })
    expect(wrapper.find('p')).toHaveStyle({ textDecoration: 'line-through' })
  })
})

