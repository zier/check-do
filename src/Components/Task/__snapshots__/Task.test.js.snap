// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task work correctly renders correctly 1`] = `
<Card
  bordered={true}
  style={
    Object {
      "padding": "0px",
    }
  }
>
  <div
    className="control-buttons"
  >
    <Icon
      className="checkButton"
      onClick={[Function]}
      type="check-circle-o"
    />
    <Icon
      className="closeButton"
      onClick={[Function]}
      type="close"
    />
  </div>
  <h4
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    title-test
  </h4>
  <p
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    description-test
  </p>
  <Tag
    className="taskTag"
    closable={false}
    key="home"
    prefixCls="ant-tag"
  >
    home
  </Tag>
</Card>
`;
